pkg/operator/controller/canary/route.go

```diff
@@ -75,6 +77,21 @@ func (r *reconciler) updateCanaryRoute(current, desired *routev1.Route) (bool, e
 		return false, nil
 	}
 
+	if len(updated.Spec.Host) == 0 && len(current.Spec.Host) != 0 {
```

Comment:
It looks like we also need to fix the function `canaryRouteChanged` to return `true` if the `Spec.Host` changed, otherwise we never get here.

/hold

Reply:
Update: I was mistaken on this.
/unhold

Reply:
https://github.com/openshift/cluster-ingress-operator/pull/965/files hasn't been fully backported yet and I was looking in the wrong branch.